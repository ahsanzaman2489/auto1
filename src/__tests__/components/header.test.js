import React from 'react';
import HeaderComponent from '../../components/header';
import {shallow} from 'enzyme';
import renderer from "react-test-renderer";
import {BrowserRouter as Router} from 'react-router-dom';

const setup = (props = {}, state = null) => {
    const defaultProps = {src: "logo.png"};
    const setUpProps = {...defaultProps, ...props};
    const wrapper = shallow(<HeaderComponent {...setUpProps}/>);
    const tree = () => renderer.create(<Router><HeaderComponent {...setUpProps}/></Router>).toJSON();
    if (state) wrapper.setState(state);
    return {wrapper, props: setUpProps, tree};
};

describe("Header component", () => {
    it('should match with snapshot', () => {
        const {tree} = setup();
        expect(tree()).toMatchSnapshot();
    });

    it('Should contain logo source', () => {
        const {wrapper, props} = setup();
        expect(wrapper.find("img").prop('src')).toBe(props.src);
    });

    it('Should render navigation list', () => {
        const {wrapper} = setup();
        expect(wrapper.find("li").length).toBeGreaterThan(0);
    });
});


