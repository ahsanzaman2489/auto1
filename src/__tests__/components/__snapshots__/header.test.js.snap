// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match with snapshot 1`] = `
<div
  className="container"
>
  <header>
    <div
      className="logo"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <img
          alt="logo"
          className="logo"
          src="logo.png"
        />
      </a>
    </div>
    <div
      className="main-menu"
    >
      <ul>
        <li>
          purchase
        </li>
        <li>
          my orders
        </li>
        <li>
          sell
        </li>
      </ul>
    </div>
  </header>
</div>
`;
